### DISCUSSION TITLE
image component in a tailwind css grid layout

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [vercel/next.js#37670] [Help] [answered]
I was trying to create a layout where the image is on the left and everything else on the right, but it didn't work out as expected, here is my code:
```
<div className="container w-full h-screen pt-6 ">
      <Meta title={movie.title} />
      <div className="px-3 h-screen gap-x-3 grid grid-cols-[6fr_4fr] grid-rows-[2fr_4fr_1fr_1fr_3fr] content-start">
        <Image
          src={`https://image.tmdb.org/t/p/original${movie.backdrop_path}`}
          width={800}
          height={480}
          className="rounded-md col-start-1 mr-3 row-span-5 "
          alt={movie.title}
        />
        <h1 className="font-bold text-2xl mt-2 col-start-2">{movie.title}</h1>
        <p className="text-gray-600 text-md mt-4 col-start-2">
          {movie.overview}
        </p>
        <p className="mt-5 text-gray-600 text-sm col-start-2">
          Genres:{' '}
          <span className="font-bold">
            {movie.genres.map((genre) => genre.name).join(', ')}
          </span>
        </p>
        <p className="text-gray-600 text-sm col-start-2">
          Release Date: <span className="font-bold">{movie.release_date}</span>
        </p>
      </div>
    </div>
```
the result:
![image](https://user-images.githubusercontent.com/61914365/173382953-08656de4-e44c-4a9e-8381-dc3b4906dc04.png)
it seems like the image is only taking up one row even though I am using `row-span-5` to tell it to span across all 5 rows, also the aspect ratio is off

**Comment by icyJoseph** [comment, marked as answer]
I think you have to wrap the Image with a `div`, and make that `div` `relative`, apply the grid classes, and margins to it, then pass a `layout="fill"` prop to the `next/image` - though I am not sure about the aspect ratio you'll get from that.

```jsx

 <div className="container h-screen w-full pt-6 ">
      <div className="grid h-screen grid-cols-[6fr_4fr] grid-rows-[2fr_4fr_1fr_1fr_3fr] content-start gap-x-3 px-3">
        <div className="relative col-start-1 row-span-5 mr-3">
          <Image
            src="/flames.jpg"
            width={800}
            height={480}
            className="rounded-md"
            layout="fill"
            alt="title"
          />
        </div>
        <h1 className="col-start-2 mt-2 text-2xl font-bold">title</h1>
        <p className="text-md col-start-2 mt-4 text-gray-600">
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. Voluptas
          nobis libero sed, illo dolores inventore aspernatur optio amet
          officiis nam voluptatem saepe sapiente ratione culpa animi. Sed,
          dolorum. Explicabo, voluptas?
        </p>
        <p className="col-start-2 mt-5 text-sm text-gray-600">
          Genres:{' '}
          {/* <span className="font-bold">
            {movie.genres.map((genre) => genre.name).join(', ')}
          </span> */}
        </p>
        <p className="col-start-2 text-sm text-gray-600">
          Release Date: <span className="font-bold">date</span>
        </p>
      </div>
    </div>
```

Do remember that `next/image` doesn't just render an `img` tag but a bunch of tags aimed at providing features the creators considered to be the best practices available as of today. So there's a span serving as top level delimiter of the image, and most often we have to wrap this bunch of tags with a wrapper to make it work within our layouts.

**Reply by Chandler-Zhu** [reply]
Thanks for your suggestion. How do you add syntax highlighting in the code block ?

**Reply by icyJoseph** [reply]
Did it work?

And for syntax highlighting start the code block with "\```css" for example for rust code:  \```rs or \```rust

```rs
let list = vec![];
```

Or for React with TypeScript use tsx

```tsx
const Hi = () => <div>Hi</div>
```

etc...  as usual, please refer to the docs https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by Chandler-Zhu** [reply]
I changed the `layout` attribute to 'responsive' now it is working
```jsx
 <div className="[grid-row:2/7] col-start-1 mr-3 rounded-md">
         <Image
            src={`https://image.tmdb.org/t/p/original${movie.backdrop_path}`}
            width="1000"
            height="600"
            layout="responsive"
            objectFit="cover"
            alt={movie.title}
          />
</div>
```

**Reply by icyJoseph** [reply]
If you think my answer was enough please consider taking it as answer, otherwise, repost your answer, explain a bit more of what you did, and label that as answer. Good stuff.
